rc.ml

```diff
@@ -0,0 +1,34 @@
+let nonopt x a b =
+  if x >= 0 && x <= a then a else b
+
+let opt x b =
+  let a = 10000 in
+  if x >= 0 && x <= a then a else b
+
+let opt_rev x b =
+  let a = 10000 in
+  if x <= a && x >= 0 then a else b
+
+let opt_l x b =
+  let a = 10000 in
+  if 0 <= x && a >= x then a else b
+
+let opt_l_rev x b =
+  let a = 10000 in
+  if a >= x && 0 <= x then a else b
+
+let opt_strict x b =
+  let a = 10000 in
+  if x >= 0 && x < a then a else b
+
+let opt_strict_rev x b =
+  let a = 10000 in
+  if x < a && x >= 0 then a else b
+
+let opt_strict_l x b =
+  let a = 10000 in
+  if 0 <= x && a > x then a else b
+
+let opt_strict_l_rev x b =
+  let a = 10000 in
+  if a > x && 0 <= x then a else b
```

Comment:
Considering my comment above concerning correctness, it'd be nice to have this as a proper test, including comparison of Int64s (and others) between `1` and `1000`, with runtime assertions, to at least check that the semantics are preserved.